### ISSUE TITLE
Merlin fails to type-check files that use the metapp extension

### ISSUE THREAD

**Issue by kiranandcode** [ocamllibs/metapp#5]
When attempting to run merlin on a file that uses the metapp extension, merlin complains that the `%meta` extension is uninterpreted and reports the error:

```
Unable to compile preprocessor: command-line\n\"'ocamlfind' 'ocamlopt' '-package' 'metapp.preutils,metapp.api,ppxlib' '-I'\n'+compiler-libs' '-w' '-40' '-shared' '/tmp/metappc806da.ml' '-o'\n'/tmp/metappc806da.cmxs'\"\nfailed with exit-code 2
```
The project otherwise is able to build correctly, and the meta-extension behaves as I expect.

For reference, this is the contents of the file on which merlin complains:
```ocaml
(* help.ml *)
let x = [%meta if Sys.ocaml_version >= "4.08.0" then [%e true] else [%e false]]
```
If I print out the value of `x` in another file, then things work as expected:
```
[@@@warning "-34-38"]
open Containers


let () =
  Format.printf "x is %b\n@." Help.x;
(* prints true *) 
```

Running on OCaml 4.14.0, merlin version v4.5-414

**Comment by thierry-martinez** [comment]
Thank you very much for reporting this! I should admit I noticed such errors quite often, but I did not take care about them because `merlin` is still quite magical for me. I just understood the problem and tried to make a fix (#7): this is not completely satisfactory, but if it solves your case, I will put it in the next release. Thank you again!

**Comment by kiranandcode** [comment]
Perfect! Thanks for the prompt response, I cloned the branch and tried it out: this is great - merlin works again with files that use the metapp extension and there are no errors!

Thanks again!
